src/ray/raylet/scheduling/local_resource_manager.cc

```diff
@@ -361,6 +365,52 @@ double LocalResourceManager::GetLocalAvailableCpus() const {
   return local_resources_.available.Sum(ResourceID::CPU()).Double();
 }
 
+std::optional<syncer::RaySyncMessage> LocalResourceManager::CreateSyncMessage(
+    int64_t after_version, syncer::MessageType message_type) const {
+  RAY_CHECK(message_type == syncer::MessageType::RESOURCE_MANAGER);
+  // We check the memory inside version, so version is not a const function.
+  // Ideally, we need to move the memory check somewhere else.
+  // TODO(iycheng): Make version as a const function.
+  auto curr_version = const_cast<LocalResourceManager *>(this)->Version();
+  if (curr_version <= after_version) {
+    return std::nullopt;
```

Comment:
when can this happen?

Reply:
When there is no new version of the message is generated. We poll for every 100ms for now.